### PR TITLE
mount: Fixes for findmount output causing idempotency issues

### PR DESCRIPTION
**Pull request by antima-gupta**
Signed-off-by: antima-gupta <agupta@msystechnologies.com>

## Description
- Fixes for CIFS mount not idempotent issue.
- Added spec for network device found as mounted.
- Removed '/' from end of the device value.

## Related Issue
#10453 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/provider/mount/linux.rb [hunk 1]
```diff
@@ -53,6 +53,10 @@ def mounted?
             when %r{\A#{Regexp.escape(real_mount_point)}\s+([/\w])+\[#{device_mount_regex}\]\s}
               mounted = true
               logger.trace("Bind device #{device_logstring} mounted as #{real_mount_point}")
+            # Permalink for network device mounted to an existing mount point: https://rubular.com/r/HicK63SqchPLvk
+            when %r{\A#{Regexp.escape(real_mount_point)}\s+#{device_mount_regex}\[([\/\w])+\]\s}
```
Comment (lamont-granquist) [inline lib/chef/provider/mount/linux.rb:57]: maybe just this?

`%r{\A#{Regexp.escape(real_mount_point)}\s+#{device_mount_regex}\[}`

does any of the stuff after the bracket really matter in terms of determining a match or not?

**Comment by lamont-granquist** [comment]
seems fine either way though, but the regexp may be overly complicated

File: lib/chef/provider/mount/linux.rb (hunk 1 above)
Comment (lamont-granquist) [reply lib/chef/provider/mount/linux.rb:57]: the extra matching on the allowed characters within the brackets may actually be brittle as well, if there's ever a mountpoint in the world which doesn't match those characters there.

Comment (antima-gupta) [reply lib/chef/provider/mount/linux.rb]: Agreed, I have updated the Regex.

**Comment by antima-gupta** [comment]
@lamont-granquist I have updated the PR, please have a look.

File: lib/chef/resource/mount.rb [hunk 2]
```diff
@@ -65,7 +66,7 @@ class Mount < Chef::Resource
 
       property :options, [Array, String, nil],
         description: "An array or comma separated list of options for the mount.",
-        coerce: proc { |arg| arg.is_a?(String) ? arg.split(",") : arg },
+        coerce: proc { |arg| mount_options(arg) }, # Please see #mount_options method.
```
Comment (tas50) [inline lib/chef/resource/mount.rb:69]: Any time we do a coerce we should make sure to add specs to test the various scenarios

Comment (tas50) [reply lib/chef/resource/mount.rb:69]: I added those here so we don't block on merging this: https://github.com/chef/chef/pull/10625

Comment (antima-gupta) [reply lib/chef/resource/mount.rb:69]: > Any time we do a coerce we should make sure to add specs to test the various scenarios

@tas50 Okay, I'll do that.

**Pull request merged**